### PR TITLE
Cleanup Upgrade/Installer Code

### LINKED ISSUE #4118: Remove Unused Upgrade Code
## Description of problem
DNN Platform upon upgrade at times will need to execute manual steps to perform the upgrade.  This is all done as part of the installer component code.  However, over time, this has resulted in the code growing substantially with each reelese.

## Description of solution
For the past 5+ years an [incremental upgrade pathway](https://dnndocs.com/content/getting-started/setup/upgrades/suggested-upgrade-path/index.html) has been suggested for upgrade.  Following this upgrade path users would do multiple upgrades PRIOR to upgrading to 9.x.

As such, in the 9.x platform, any manual upgrade steps that exist prior to 9.3.2 would be un-used as the prior 9.3.2 upgrade path would have already executed those steps.  

To be safe, and considering that users do not always follow the upgrade path, I propose the removal of all manual upgrades steps prior to 7.4.2.  This will support 4 individual hops back from the current recommendations.  new installations are NOT impacted by this code. 

## Description of alternatives considered
We could leave this code, but the amount of bloat and maintenance benefit is huge.

### PR DESCRIPTION
**Pull request by mitchelsellers**
Resolves #4118 

This PR removes all unused code that is used to complete upgrades prior to 7.4.2 and earlier.  

## Background
A bit of important background, in addition to what was initially noted in the Issue.

DNN Platform, upon install starts by applying an initial script, then it "Upgrades" the application from one version to another until it is fully updated.  Currently, the [DotNetNuke.Install.Config](https://github.com/dnnsoftware/Dnn.Platform/blob/develop/DNN%20Platform/Website/Install/DotNetNuke.install.config.resources) file starts this process at 08.00.04 and upgrades from there.  For extra safety I only removed upgrade steps to 7.4.2.

As a reminder, if you follow the upgrade path, the removal of these elements will have NO impact as the upgrade to 7.4.2 alone would have processed these items.  Given this, I do NOT believe that this a breaking change.

## Included Cleanups

Included in this cleanup is removal of the following items:

* Internal elements fully contained inside of the Upgrade.cs
* Removal of XML Configuraiton Merge elements no longer needed
* Documentation of a few API's that are no longer used by the platform, however, were exposed publicly in error

### CODE REVIEW

**Review by bdukes — APPROVED** [review]

File: DNN Platform/Library/Services/Installer/LegacyUtil.cs [hunk 1]
```diff
@@ -343,77 +345,6 @@ public static void ProcessLegacyModule(DesktopModuleInfo desktopModule)
             }
         }
 
-        public static void ProcessLegacyModules()
```
[the remark this answers is not in the document]
Comment (mitchelsellers) [reply DNN Platform/Library/Services/Installer/LegacyUtil.cs:346]: Not exactly, ore of a difference in approach after seeing all of the internal usages.  The more I think about it, the more I'm in favor of just removing the others.  

Yes they are public, but they are internal utilities for the installer and unused.

File: DNN Platform/Library/Common/Initialize.cs [hunk 2]
```diff
@@ -455,7 +455,6 @@ private static string InitializeApp(HttpApplication app, ref bool initialized)
                     Globals.DatabaseEngineVersion = GetDatabaseEngineVersion();
 
                     // Try and Upgrade to Current Framewok
```
Comment (valadas) [inline DNN Platform/Library/Common/Initialize.cs:457]: Looks good to me except this little stray comment that is no longer relevant

```suggestion
                    Globals.DatabaseEngineVersion = GetDatabaseEngineVersion();
```

**Review by valadas — CHANGES_REQUESTED** [review]

File: DNN Platform/Library/Services/Installer/LegacyUtil.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Library/Services/Installer/LegacyUtil.cs:346]: I agree with @mitchelsellers here, although this was public, it really is something used by Dnn modules install/upgrades and it relates more to legacy manifests than developers actually calling this API. I vote for removing it for that reason.

**Review by valadas — APPROVED** [review]
Looks good to me

**Pull request merged**